File: Algorithms3/LastLections/n1.py
with open('input.txt') as f:
    s = f.read()
d = {}
max = 0
for el in s:
    if el != ' ' and el != '\n':
        if el not in d:
            d[el] = 0
        d[el] += 1
        if d[el] > max:
            max = d[el]
m = sorted(d)
while max > 0:
    for el in m:
        if d[el] == max:
            print('#', end='')
            d[el] -= 1
        else:
            print(' ', end='')
    print()
    max -= 1
for el in m:
    print(el, end='')
print()

File: Algorithms1/Lection8/n1.cpp
#include <iostream>
#include <fstream>

struct node {
    int data;
    struct node * left = nullptr;
    struct node * right = nullptr;
};

int insert(struct node * &root, int data) {
    if (root == nullptr) {
        root = new struct node;
        root->data = data;
        return 1;
    } else if (data > root->data) {
        return insert(root->right, data) + 1;
    } else if (data < root->data) {
        return insert(root->left, data) + 1;
    }
    return -1;
}

int main() {
    std::ifstream file("input.txt");
    struct node * root = nullptr;
    int c, max;
    file >> c;
    max = insert(root, c);
    for (file >> c; !file.eof() && c != 0; file >> c) {
        int max_tmp = insert(root, c);
        if (max_tmp > max) {
            max = max_tmp;
        }
    }
    std::cout << max << std::endl;
    return 0;
}